src/commands/integrations/flows/test.ts

```diff
@@ -1,160 +1,382 @@
+    ux.action.stop();
+
+    this.startTime = Date.now();
+
+    const flagString = `${payloadFilePath ? `-p=${payloadFilePath} ` : ""}${
+      tailLogs ? "--tail-logs " : ""
+    }${tailStepResults ? "--tail-results " : ""}${sync ? "--sync " : ""}${
+      autoEndPoll ? "--cni-auto-end " : ""
+    }${resultFilePath ? `-r=${resultFilePath} ` : ""}`;
+
+    this.log(`
+To re-run this flow directly:
+prism cni:test:flow -u=${invokeUrl} ${flagString}
+`);
+
+    const executionId = response.headers["prismatic-executionid"];
+
+    if (!response.data.executionId) {
+      // Log execution ID's separately for synchronously-run flows.
+      this.log(`Execution ID: ${executionId}\n`);
     }
+
+    // Log the results.
+    this.log(`${JSON.stringify(response.data)}\n`);
+
+    if (!(tailLogs || tailStepResults)) return;
+
+    // If tailing logs or step results, show relevant messaging & setup polling promises.
+    this.warn(
+      "While the timestamps are accurate, logs & step results may not arrive in chronological order.",
+    );
+    this.log(
+      `\nPress CMD+C/CTRL+C to stop polling. ${
+        autoEndPoll ? "" : "This process will timeout after 20 minutes."
+      }\n`,
+    );
+
+    if (resultFilePath) {
+      await fs.appendFile(resultFilePath, "timestamp,type,data\n");
+    }
+
+    const tailPromises = [];
+    this.startTime = Date.now();
+
+    if (tailLogs) tailPromises.push(this.tailLogs(executionId));
+    if (tailStepResults) tailPromises.push(this.tailStepResults(executionId));
+
+    await Promise.all(tailPromises);
   }
 
   private async tailLogs(executionId: string) {
-    const { flags } = await this.parse(TestCommand);
+    const {
+      flags: { "cni-auto-end": autoEndPoll, "result-file": resultFilePath, timeout },
+    } = await this.parse(TestFlowCommand);
 
     let nextCursor: string | undefined = undefined;
+
     while (true) {
-      await ux.wait(500);
+      await ux.wait(this.getPollIntervalMs());
 
       const result: any = await this.fetchLogs(executionId, nextCursor);
       if (result === undefined) continue;
 
-      const { logs, cursor, executionComplete } = result;
-
+      const { logs, cursor } = result;
       nextCursor = cursor;
 
       ux.table(
         logs,
         {
           timestamp: {},
           severity: {
-            minWidth: 12,
+            minWidth: 15,
           },
           message: {},
         },
-        { ...flags, "no-header": true },
+        {
+          "no-header": true,
+        },
+      );
+
+      if (resultFilePath) {
+        for (const log of logs) {
+          await fs.appendFile(resultFilePath, `${log.timestamp},${log.severity},${log.message}\n`);
+        }
+      }
+
+      if (await this.shouldEnd(executionId, autoEndPoll, timeout)) {
+        return;
+      }
+    }
+  }
+
+  private async tailStepResults(executionId: string) {
+    const {
+      flags: { "cni-auto-end": autoEndPoll, "result-file": resultFilePath, timeout },
```

Comment:
```suggestion
      flags: { 
        "cni-auto-end": autoEndPollFlag, 
        "result-file": resultFilePathFlag, 
        timeout: timeoutFlag 
      },
```
I saw you do this earlier, and really liked it. Thoughts on postfixing these flags with `Flag` to create a more concise binding convention?

Reply:
I'd prefer not to, because then the user will have to enter their flags like `prism integrations:flow:test --result-file-flag=file.txt --timeout-flag=100`, etc.